### PR TITLE
[KYUUBI #3848] [Improvement] Sort config map returned in`KyuubiConf.getAll`

### LINKED ISSUE #3848: [Improvement] Sort config map returned in `KyuubiConf.getAll`
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Sort the config map returned in `KyuubiConf.getAll` to improve readability of engine launching configs in output.

Snapshot of logs of engine launching commands, which is in random order and hard to read or lookup the relevant config in series.
image

### How should we improve?

_No response_

### Are you willing to submit PR?

- [X] Yes. I can submit a PR independently to improve.
- [ ] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

to close #3848 .

Sort the config map returned in KyuubiConf.getAll to improve readability of engine launching configs in output.

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by bowenliang123** [comment]
cc @pan3793

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 1]
```diff
@@ -114,7 +115,7 @@ case class KyuubiConf(loadSysDefault: Boolean = true) extends Logging {
 
   /** Get all parameters as map */
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:116]: Please update the comment to mention the ordered. And better to add a simple UT to ensure it works as expected.

Comment (bowenliang123) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala]: UT added and method comment updated.

**Comment by pan3793** [comment]
Thanks, merging to master

**Pull request closed without merging**